cassandra/components.py

```diff
@@ -1161,3 +1160,390 @@ def run_component(self):
+
+        :param raise_msg:               Optional. Raise exception if True; default True
+        :type raise_msg:                bool
+
+        """
+
+        if log_msg:
+            logging.error(message)
+
+        if raise_msg:
+            raise exception(message)
+
+    def validate_params(self):
+        """Ensure params are present for this component."""
+
+        # list of expected params
+        param_list = [self.RDS_FILE_FIELD, self.CLIMATE_VAR_NAME_FIELD, self.SCENARIO_FIELD, self.UNITS_FIELD,
+                      self.START_YEAR_FIELD, self.THROUGH_YEAR_FIELD]
+
+        for i in param_list:
+
+            if i not in self.params:
+                msg = f"{self.__class__} Required parameter '{i}' not in config file."
+                self.log_raise_exception(KeyError, msg)
+
+    def validate_year(self, yr, min_yr=0, max_yr=10000):
+        """Ensure years are within a reasonable range and are integers.
+
+        :param yr:                      Target year
+        :type yr:                       int
+
+        :param min_yr:                  Minimum year allowable
+        :type min_yr:                   int
+
+        :param max_yr:                  Maximum year allowable
+        :type max_yr:                   int
+
+        :return:                        [0] validated start year as a four digit integer
+                                        [1] validated end year as a four digit integer
+
+        """
+
+        # validate that the year can be converted to an integer
+        valid_yr = self.validate_int(yr)
+
+        if (valid_yr < min_yr) or (valid_yr > max_yr):
+            msg = f"{self.__class__} Year '{valid_yr}' is outside of the reasonable bounds [{min_yr} - {max_yr}]."
+            self.log_raise_exception(ValueError, msg)
+
+        else:
+            return valid_yr
+
+    def validate_int(self, value):
+        """Ensure the value can be converted to an integer.
+
+        :param value:                   Target value
+        :type value:                    int, float, str
+
+        :return:                        Validated int of value
+
+        """
+
+        if type(value) is int:
+            return value
+
+        try:
+            return int(value)
+
+        except ValueError:
+            msg = f"{self.__class__} Value '{value}' not able to be converted to an integer as expected."
+            self.log_raise_exception(ValueError, msg)
+
+    def validate_file_exist(self, file_path):
+        """Ensure file exists.
+
+        :param file_path:               Full path with file name an extension to in input file.
+        :type file_path:                str
+
+        :return:                        Validated file path
+
+        """
+
+        if os.path.isfile(file_path):
+            return file_path
+
+        else:
+            msg = f"Input file '{file_path}' does not exist."
+            self.log_raise_exception(FileNotFoundError, msg)
+
+    def rds_to_dict(self):
```

Comment:
I think we can get away without this function, if we create an exported function in fldgen to extract the information we need and return it as a simple vector.